### PR TITLE
feat: add secretsReader

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: src/main/java/nva/commons/utils/aws/SecretsReader.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import nva.commons.utils.JsonUtils;
+import nva.commons.utils.attempt.Failure;
+import nva.commons.utils.attempt.Try;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SecretsReader {
+
+    private static final Logger logger = LoggerFactory.getLogger(SecretsReader.class);
+    public static final String COULD_NOT_READ_SECRET_ERROR = "Could not read secret:";
```
Comment (janbj) [inline src/main/java/nva/commons/utils/aws/SecretsReader.java:22]: Maybe add a space at the end here

File: src/test/java/nva/commons/utils/aws/SecretsReaderTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        return providedSecretName;
+    }
+
+    private GetSecretValueResult createGetSecretValueResult(String secretString) {
+        return new GetSecretValueResult()
+            .withSecretString(secretString)
+            .withName(SECRET_NAME);
+    }
+
+    private String createSecretObject() throws JsonProcessingException {
```
Comment (janbj) [inline src/test/java/nva/commons/utils/aws/SecretsReaderTest.java:100]: createSecretJsonObject?

File: src/test/java/nva/commons/utils/aws/SecretsReaderTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        String providedSecretName = parseRequest(invocation);
+        if (providedSecretName.equals(SECRET_NAME)) {
+            String secretString = createSecretObject();
+            return createGetSecretValueResult(secretString);
+        } else {
+            throw new ResourceNotFoundException(ERROR_MESSAGE_FROM_AWS_SECRET_MANAGER);
+        }
+    }
+
+    private String parseRequest(InvocationOnMock invocation) {
```
Comment (janbj) [inline src/test/java/nva/commons/utils/aws/SecretsReaderTest.java:88]: getSecretNameFromRequest?

File: src/test/java/nva/commons/utils/aws/SecretsReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        assertThat(value, is(equalTo(SECRET_VALUE)));
+    }
+
+    private SecretsReader createSecretsReader() {
+        AWSSecretsManager secretsManager = mock(AWSSecretsManager.class);
+        when(secretsManager.getSecretValue(any(GetSecretValueRequest.class))).thenAnswer(this::provideSecret);
+        return new SecretsReader(secretsManager);
+    }
+
+    private GetSecretValueResult provideSecret(InvocationOnMock invocation) throws JsonProcessingException {
```
Comment (janbj) [inline src/test/java/nva/commons/utils/aws/SecretsReaderTest.java:78]: createGetSecretValueResult?

File: src/test/java/nva/commons/utils/aws/SecretsReaderTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        assertThat(exception.getMessage(), is(equalTo(ForbiddenException.DEFAULT_MESSAGE)));
+    }
+
+    @Test
+    public void fetchSecretReturnsSecretValueWhenSecretNameAndSecretKeyAreCorrect() throws ForbiddenException {
+        String value = secretsReader.fetchSecret(SECRET_NAME, SECRET_KEY);
+        assertThat(value, is(equalTo(SECRET_VALUE)));
+    }
+
+    private SecretsReader createSecretsReader() {
```
Comment (janbj) [inline src/test/java/nva/commons/utils/aws/SecretsReaderTest.java:72]: createSecretsReaderMock?

File: src/test/java/nva/commons/utils/aws/SecretsReaderTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    public static final String SECRET_VALUE = "SECRET_VALUE";
+    public static final String SECRET_KEY = "SECRET_KEY";
+    public static final String SECRET_NAME = "SECRET_NAME";
+    public static final String WRONG_SECRET_NAME = "WRONG_SECRET_NAME";
+    public static final String WRONG_SECRET_KEY = "WRONG_KEY";
+    public static final String ERROR_MESSAGE_FROM_AWS_SECRET_MANAGER = "Secret not found";
+    private final SecretsReader secretsReader;
+
+    public SecretsReaderTest() throws JsonProcessingException {
```
Comment (janbj) [inline src/test/java/nva/commons/utils/aws/SecretsReaderTest.java:36]: Should this be done as an @BeforeEach annotated init method?

Comment (axthosarouris) [reply src/test/java/nva/commons/utils/aws/SecretsReaderTest.java:36]: Not necessary. It needs to be run only once.

**Pull request merged**